app/src/main/java/com/worldsnas/starplayer/ExoPlayerService.kt

```diff
@@ -0,0 +1,387 @@
+package com.worldsnas.starplayer
+
+import android.app.Notification
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.os.IBinder
+import android.text.TextUtils
+import androidx.core.app.NotificationCompat
+import com.google.android.exoplayer2.*
+import com.google.android.exoplayer2.source.*
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray
+import com.google.android.exoplayer2.upstream.DefaultHttpDataSourceFactory
+import com.worldsnas.starplayer.model.Music
+import java.util.*
+
+class ExoPlayerService : Service(), Player.EventListener {
+
+
+    lateinit var mediaSource: ConcatenatingMediaSource
+
+
+    lateinit var player: SimpleExoPlayer
+
+    private var playbackPosition: Long = 0
+    private lateinit var musicList: ArrayList<Music>
+    private var currentMusicPosition: Int = 0
+
+
+    override fun onDestroy() {
+        super.onDestroy()
+        releasePlayer()
+        releasePlayer()
+    }
+
+
+    private fun play(trackModel: Music) {
+        musicList = ArrayList<Music>()
+        musicList.add(trackModel)
+        play(musicList)
+    }
+
+    private fun play(trackModels: ArrayList<Music>) {
+        playbackPosition = 0
```

Comment:
behtar nist ghabl az pakhsh yedor state ghabli ro chek konim?
on media source hayi ke az ghabl hast ro pak ya haz konim (nemidonam aslan niaz has ya na)

Reply:
in method daghighan bad az ejra shodan service va initialize shodan player call mishe.
fekr nakonam niaz bashe vali mitooniam check konim ke age player dar hale pakhsh bood yebar release she va 2bare initialize she